### ISSUE TITLE
Handle 2nde délib ?

### ISSUE THREAD

**Issue by RouxRC** [regardscitoyens/the-law-factory-parser#63]
techniquement il devrait y avoir deux versions du texte à une même étape d'hémicycle dans certains cas : les secondes délibérations
L'amdmt http://www.assemblee-nationale.fr/14/amendements/0044S/1.asp porte sur un article 4 d'un texte qui n'a que 3 articles, parce que c une seconde délib, après qu'en lecture en hémicycle ils aient adopté un amendement ajoutant un article après l'article 3 http://www.assemblee-nationale.fr/14/amendements/0044/AN/2.asp

**Comment by promethe42** [comment]
En fait "APRÈS ART. 3" ça veut dire créer un article 4 ?

https://www.nosdeputes.fr/14/amendements/44/xml

Comment on fait pour avoir la bonne numérotation quand ce cas existe avec plusieurs amendements ?

**Comment by njoyard** [comment]
Non, la numérotation originale est toujours (il me semble) conservée. On créera alors un 3 bis, un 3 ter, etc. puis s'il le faut un 3 bis A, 3 bis B, etc.

Je n'arrive pas cependant à trouver une logique systématique. Parfois, "après art. 3" aboutit directement à un 3 A sans passer par un 3 bis...

**Comment by RouxRC** [comment]
APES ART ou AVANT ART veut dire créer un nouvel article avant ou après. Si plusieurs sont créés, c'est le service dit de la séance qui va décider de l'ordre dans lequel ils sont discutés et donc insérés entre les deux, mais il n'y a pas vraiment de moyen sûr de savoir a priori dans quel ordre les insérer sans disposer du texte dans sa version a posteriori.

Dans le cas présent, le texte ne faisant que 3 articles, les articles ajoutés après l'article 3 peuvent être simplement incrémentés et en l'occurence l'amendement créait donc bien un article 4.

Mais pour répondre à @njoyard on ajoute bis, ter etc quand on ajoute après, et on ajoute A B etc quand on ajoute avant un bis ter etc. 3A donc vient après 3 et avant 3 bis.
cf exemple d'ordonnancement géré ici : https://github.com/regardscitoyens/the-law-factory-parser/blob/master/scripts/collectdata/sort_articles.py#L126-L160

**Comment by RouxRC** [comment]
À noter également qu'à la toute dernière étape, les services renumérotent, c'est-à-dire que les articles bis ter ABD etc disparaissent tous pour une numérotation canonique de 1 à N sans trou. Mais a priori il ne devrait pas y avoir d'amendement possible sur ces versions là du texte pusique ce sont justement les versions finales.

**Comment by promethe42** [comment]
Bon... c'est un joyeux b***el niveau traçabilité donc.
Ca confirme mon intuition de faire un "hash" qui représente la modification apportée par le texte, pour pouvoir comparer les articles en comparant les hash. Si deux articles ont le même hash mais pas le même numéro, c'est soit un doublon soit un renommage.

On est pas rendu...
